app/src/app/components/Toolbar/ZonePicker.tsx

```diff
@@ -13,7 +14,7 @@ export function ZonePicker() {
   };
 
   return (
-    <div>
+    <div style={access === 'read' ? {pointerEvents: 'none', opacity: 0.5} : {}}>
```

Comment:
Medium: This could be refactored since it appears in multiple places

```typescript
const AccessBoundary: React.FC<{children: React.ReactNode}> = ({children}) => {
  const access = useMapStore(state => state.mapStatus?.access)
  return <Box className={access === 'read' && 'pointer-events-none opacity-50'}>{children}</Box>
}

```